cw_decred/lib/wallet_addresses.dart

```diff
@@ -7,13 +7,27 @@ import 'package:cw_decred/api/libdcrwallet.dart' as libdcrwallet;
 class DecredWalletAddresses extends WalletAddresses {
   DecredWalletAddresses(WalletInfo walletInfo) : super(walletInfo);
 
+  String currentAddr = '';
+
   @override
   String get address {
-    return libdcrwallet.currentReceiveAddress(walletInfo.name) ?? '';
+    final cAddr = libdcrwallet.currentReceiveAddress(walletInfo.name) ?? '';
+    if (cAddr != '') {
+      currentAddr = cAddr;
+    }
+    return currentAddr;
   }
 
   String generateNewAddress() {
-    return libdcrwallet.newExternalAddress(walletInfo.name) ?? '';
+    final nAddr = libdcrwallet.newExternalAddress(walletInfo.name) ?? '';
+    if (nAddr != '') {
+      currentAddr = nAddr;
+    }
+    return nAddr;
+  }
+
+  String get currentAddress {
+    return currentAddr;
   }
```

Comment:
Is this getter used outside decred lib?

Reply:
No. Was there a better way of doing this? Can I just access the fields?

Reply:
Yeah looks like I can so changed it https://github.com/JoeGruffins/cake_wallet/compare/bd6be939d67b6733abce0f9dbc8a72dbc5e99d9e..bf1900c96c46e2ebb44b1358daf7d98661053171